howtos/tech/build-cmocka.rst

```diff
@@ -0,0 +1,135 @@
+.. _build-cmocka:
+
+Build Cmocka for Xtensa
+#######################
+
+Cmocka for SOF is built automatically by default.
+However sometimes you may need to have prebuilt version that can be used with :ref:`CMOCKA_DIRECTORY <cmocka-directory-label>`.
+
+This article exaplains how to build Cmocka manually.
+
+Cmocka fork
+***********
+
+We use our Cmocka fork that adds some options for embedded compilers:
+
+WITH_POSITION_INDEPENDENT_CODE
+   Some compilers cannot compile CMake's test program with PIC,
+   so you can disable it with this option.
+
+WITH_TINY_CONFIG
+   Usually compiler checks are cheap, so configuration is quick.
+   However many compilers are expensive to call,
+   so this option can be used to minimize checks count.
+
+WITH_SHARED_LIB
+   For compilers that cannot build shared libs and still want to
+   use make / make install for making prebuilt libs
+
+All changes made to Cmocka are enabled with these options, without them it works just like vanilla Cmocka.
+
+Clone repo and enter its directory, because all examples will be executed there:
+
+   .. code-block:: bash
```

Comment:
Line 33: This applies to all of the code blocks. There's no need to indent the code block directive. It's getting interpreted as a code block inside of a block quote.

Reply:
Thanks for noticing